src/openzeppelin/token/erc20/interface.cairo

```diff
@@ -1,45 +1,77 @@
 use starknet::ContractAddress;
 
-#[abi]
-trait IERC20 {
-    #[view]
-    fn name() -> felt252;
-    #[view]
-    fn symbol() -> felt252;
-    #[view]
-    fn decimals() -> u8;
-    #[view]
-    fn total_supply() -> u256;
-    #[view]
-    fn balance_of(account: ContractAddress) -> u256;
-    #[view]
-    fn allowance(owner: ContractAddress, spender: ContractAddress) -> u256;
-    #[external]
-    fn transfer(recipient: ContractAddress, amount: u256) -> bool;
-    #[external]
-    fn transfer_from(sender: ContractAddress, recipient: ContractAddress, amount: u256) -> bool;
-    #[external]
-    fn approve(spender: ContractAddress, amount: u256) -> bool;
+#[starknet::interface]
+trait IERC20<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+    fn total_supply(self: @TState) -> u256;
+    fn balance_of(self: @TState, account: ContractAddress) -> u256;
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 
-#[abi]
-trait IERC20Camel {
-    #[view]
-    fn name() -> felt252;
-    #[view]
-    fn symbol() -> felt252;
-    #[view]
-    fn decimals() -> u8;
-    #[view]
-    fn totalSupply() -> u256;
-    #[view]
-    fn balanceOf(account: ContractAddress) -> u256;
-    #[view]
-    fn allowance(owner: ContractAddress, spender: ContractAddress) -> u256;
-    #[external]
-    fn transfer(recipient: ContractAddress, amount: u256) -> bool;
-    #[external]
-    fn transferFrom(sender: ContractAddress, recipient: ContractAddress, amount: u256) -> bool;
-    #[external]
-    fn approve(spender: ContractAddress, amount: u256) -> bool;
+#[starknet::interface]
+trait IERC20Camel<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+    fn totalSupply(self: @TState) -> u256;
+    fn balanceOf(self: @TState, account: ContractAddress) -> u256;
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transferFrom(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
+}
+
+#[starknet::interface]
+trait IERC20CamelOnly<TState> {
+    fn totalSupply(self: @TState) -> u256;
+    fn balanceOf(self: @TState, account: ContractAddress) -> u256;
+    fn transferFrom(
```

Comment:
Hmm if we're going to define the full camel interface as above, would it make sense to change this to just a trait? We're just using it to create the camel impl, no?

Reply:
I think that makes sense. I'm defining the complete Camel interface anyway, for using the agnostic methods in the dispatcher for testing (in one specific case I thought made sense). 

I will remove the attribute.